uint8 down
uint8 right
uint8 up
uint8 left

uint8 cross
uint8 circle
uint8 triangle
uint8 square

uint8 l1
uint8 l2
uint8 l3
float32 l2_analog
uint8 r1
uint8 r2
uint8 r3
float32 r2_analog

uint8 select
uint8 start
uint8 ps
uint8 trackpad

float32 left_x
float32 left_y
float32 right_x
float32 right_y

uint8 trackpad0_status
float32 trackpad0_x
float32 trackpad0_y
uint8 trackpad1_status
float32 trackpad1_x
float32 trackpad1_y